app/lib/schema_file_loader.rb

```diff
@@ -35,12 +37,14 @@ def download_and_unzip_schemas_from_s3(dest_dir)
 
     def download_schemas_from_s3(dest_dir)
       s3_client = Aws::S3::Client.new(region: REGION, credentials: s3_credentials)
-      get_missing_downloads(dest_dir).each do |download_path|
+      get_missing_downloads(dest_dir).each do |(download_path, optional)|
         s3_client.get_object(
           response_target: download_path,
           bucket: BUCKET,
           key: File.basename(download_path),
         )
+      rescue Aws::S3::Errors::NoSuchKey
+        next if optional
```

Comment:
if you `raise if not optional`, then i think you can get rid of this explicit `next` call, since it doesn't skip any subsequent behavior inside the loop